pygsti/circuits/circuit.py

```diff
@@ -1752,12 +1802,12 @@ def delete_layers(self, layers=None):
             del self._labels[i]
 
         #Shift compilable layer indices as needed
-        if len(self._compilable_layer_indices_tup) > 0:  # begins with __CMPLBL__
+        if self._compilable_layer_indices_tup:
             deleted_indices = set(layers)
-            new_inds = list(filter(lambda x: x not in deleted_indices, self._compilable_layer_indices_tup[1:]))
+            new_inds = list(filter(lambda x: x not in deleted_indices, self._compilable_layer_indices_tup))
```

Comment:
Is this ``list(filter(...))`` pattern really needed? I imagine it would suffice to do
```python
new_inds = [x for x in self._compilable_layer_indices_tup if x not in deleted_indices]
```

Reply:
Good point, much more readable (one doesn't need to remember what filter does). I've updated this in the latest commit and refactored the remaining instances of filter in this code as well.